extensions/amp-story-360/0.1/amp-story-360.js

```diff
@@ -202,22 +266,198 @@ export class AmpStory360 extends AMP.BaseElement {
+      rot
+    );
+    rot = Matrix.mul(
+      3,
+      Matrix.rotation(3, 2, 1, (Math.PI / 180.0) * e.beta),
+      rot
+    );
+    rot = Matrix.mul(
+      3,
+      Matrix.rotation(3, 0, 2, (Math.PI / 180.0) * e.gamma),
+      rot
+    );
+    this.renderer_.setCamera(rot, 1);
+    this.renderer_.render(true);
+  }
+
+  /**
+   * Creates a "activate" button to request DeviceOrientation permissions.
+   * Simulates a click event to call this method on page load.
+   * @private
+   */
+  buildPermissionButton_() {
+    const tempButton = document.createElement('div');
+    tempButton.addEventListener('click', () => {
+      if (this.win.DeviceOrientationEvent.requestPermission) {
+        this.win.DeviceOrientationEvent.requestPermission()
+          // Render activate button if permissions aren't set yet.
+          .catch(() => {
+            const activateButton = buildActivateButtonTemplate(this.element);
+            this.element.appendChild(activateButton);
+
+            activateButton.addEventListener('click', () => {
+              this.requestGyroscopePermissions_();
+            });
+          })
+          // If permissions are already set, handle permission state.
+          .then((permissionState) => {
+            permissionState && this.setPermissionState_(permissionState);
+          });
+      }
+    });
+    tempButton.click();
+  }
+
+  /**
+   * @param {string} permissionState
+   * @private
+   */
+  setPermissionState_(permissionState) {
+    if (permissionState === 'granted') {
+      this.storeService_.dispatch(Action.SET_GYROSCOPE_PERMISSION, 'granted');
+    } else if (permissionState === 'denied') {
+      this.storeService_.dispatch(Action.SET_GYROSCOPE_PERMISSION, 'denied');
+    }
+  }
+
+  /**
+   * Toggles class on amp-story to show or hide activate button.
+   * @param {boolean} hidePermissionButton
+   * @private
+   */
+  togglePermissionClass_(hidePermissionButton) {
+    this.mutateElement(() => {
+      this.element.classList.toggle(
+        'i-amphtml-story-360-hide-permissions-ui',
+        hidePermissionButton
+      );
+    });
+  }
+
+  /** @private */
+  requestGyroscopePermissions_() {
+    if (this.win.DeviceOrientationEvent.requestPermission) {
+      this.win.DeviceOrientationEvent.requestPermission()
+        .then((permissionState) => {
+          this.setPermissionState_(permissionState);
+        })
+        .catch((error) => {
+          dev().error(TAG, `Gyroscope permission error: ${error.message}`);
+        });
+    }
+  }
+
   /** @override */
   isLayoutSupported(layout) {
     return isLayoutSizeDefined(layout);
   }
 
+  /**
+   * Checks if the image is larger than the GPUs max texture size.
+   * Scales the image down if neededed.
+   * Returns the image element if image is within bounds.
+   * If image is out of bounds, returns a scaled canvas element.
+   * @param {!Element} imgEl
+   * @return {!Element}
+   * @private
+   */
+  checkImageReSize_(imgEl) {
```

Comment:
Can you create another PR with this fix? It'd be great to have Tom look into it and I wouldn't want to block this PR on it

Reply:
Yes. I'll also remove panorama1 since that image is too big and will break the experience on mobile.

I'm feeling pretty good about this fallback since the image will still be the maximum size that the GPU can handle but I'm happy to get more feedback on it!